### DISCUSSION TITLE
sshing by multiple labels

### DISCUSSION THREAD

**Discussion by nate-thirdwave** [gravitational/teleport#31029] [Q&A] [answered]
I'm trying to use `tsh ssh` to execute commands across sections of my cluster.  So I'd like to, say, execute a command on all machines with label `type=southern` AND `product=peach`.  But if I try the naive `tsh ssh type=southern product=peach <cmd>` it overselects to either just the first label, or to the additive set of things with either label set to that value.   Is there a way to run commands on the set intersection of labels, or to provide a label expression?

**Comment by philip-teleport** [comment, marked as answer]
Try this: `tsh ssh username@type=southern,product=peach <cmd>`

**Reply by webvictim** [reply]
You can also `tsh ssh -l <login> type=southern,product=peach <cmd>` if your usernames contain `@` signs, or you just want to make sure that it parses a little more reliably.

**Comment by nate-thirdwave** [comment]
That works a treat. Thanks! Worth dropping that example in the docs as well!
